src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -707,10 +707,80 @@ def algebraic_dependence(self):
         RRR = PolynomialRing(K,r,tuple(Ts),order='negdeglex')
         return RRR.ideal(JJ.gens())
 
+    def coefficients_monomials(self, order=None, sparse=True):
+        """
+        Return the matrix of coefficients ``A`` and
+        the matching vector of monomials ``v``, such that ``A*v == vector(self)``.
+
+        Thus value of ``A[i,j]`` corresponds the coefficient of the
+        monomial ``v[j]`` in the ``i``-th polynomial in this system.
+
+        Monomials are ordered w.r.t. the term ordering of ``order``
+        if given; otherwise, they are ordered w.r.t. ``self.ring()``
+        in reverse order, i.e., such that the smallest entry comes last.
+
+        INPUT:
+
+        - ``sparse`` - construct a sparse matrix (default: ``True``)
+        - ``order`` - a list or tuple specifying the order of monomials (default: ``None``)
+
+        EXAMPLES::
+
+            sage: # needs sage.libs.singular
+            sage: P.<a,b,c,d> = PolynomialRing(GF(127), 4)
+            sage: I = sage.rings.ideal.Katsura(P)
+            sage: I.gens()
+            [a + 2*b + 2*c + 2*d - 1,
+             a^2 + 2*b^2 + 2*c^2 + 2*d^2 - a,
+             2*a*b + 2*b*c + 2*c*d - b,
+             b^2 + 2*a*c + 2*b*d - c]
+            sage: F = Sequence(I)
+            sage: A,v = F.coefficients_monomials()
+            sage: A
+            [  0   0   0   0   0   0   0   0   0   1   2   2   2 126]
+            [  1   0   2   0   0   2   0   0   2 126   0   0   0   0]
+            [  0   2   0   0   2   0   0   2   0   0 126   0   0   0]
+            [  0   0   1   2   0   0   2   0   0   0   0 126   0   0]
+            sage: v
+            (a^2, a*b, b^2, a*c, b*c, c^2, b*d, c*d, d^2, a, b, c, d, 1)
+            sage: A*v
+            (a + 2*b + 2*c + 2*d - 1, a^2 + 2*b^2 + 2*c^2 + 2*d^2 - a,
+             2*a*b + 2*b*c + 2*c*d - b, b^2 + 2*a*c + 2*b*d - c)
+        """
+        from sage.modules.free_module_element import vector
+        from sage.matrix.constructor import Matrix
+
+        R = self.ring()
+        f = tuple(self)
+        nf = len(f)
+
+        if order is None:
+            m = sorted(self.monomials(), reverse=True)
+        else:
+            if isinstance(order, (list, tuple)):
+                m = order
+            else:
+                raise ValueError("order argument can only accept list or tuple.")
+        nm = len(m)
+
+        #construct dictionary for fast lookups
+        v = dict(zip(m, range(len(m))))
+
+        A = Matrix(R.base_ring(), nf, nm, sparse=sparse)
+
+        for x in range(nf):
+            poly = f[x]
+            for y in poly.monomials():
+                try:
+                    A[x, v[y]] = poly.monomial_coefficient(y)
+                except KeyError:
+                    raise ValueError("order argument does not contain all monomials")
```

Comment:
I think that this could be slightly simplified by iterating over `poly` itself:
```
for c, m in f[x]:
    try:
        A[c, v[m]] = c
    except KeyError:
        ...
```

Reply:
But then what will be x ?

Reply:
But here, A[c, v[m]] will throw an error because in the first index, instead of the coefficient (i.e., c) of a monomial, there should be an index that represents the row of the coefficient matrix.

Reply:
You are quite right, I mistook the `x` for a `c`.  However, the following might be right:
```
y = dict(zip(m, range(len(m))))
for x, poly in enumerate(f):
    for c, m in poly:
        try:
          A[x, y[m]] = c
        except KeyError:
          ...
```
Could you check?  I also renamed `v` to `y` for clarity.

Reply:
I have changed the code according to your suggestions, please review it once, and sorry of delay, I was finding a way to make A*v == self but I didn't find a way to do so. :disappointed:

Reply:
Ohh ! I just realised that below line may through an error in some cases 

```python
    for c, m in poly:
```

It will happen in case of constant terms, because there will not be enough values to unpack in poly.

Reply:
@mantepse can you please review these changes.